apiserver/facades/client/application/application.go

```diff
@@ -472,6 +472,21 @@ func (api *APIBase) deployApplication(
 		return errors.Trace(err)
 	}
 
+	// If the charm specifies a unique architecture, ensure that is set in
+	// the constraints. Charmhub handles any existing arch constraints.
+	cons := args.Constraints
+	if !cons.HasArch() {
+		arches := set.NewStrings()
+		for _, base := range ch.Manifest().Bases {
+			for _, arch := range base.Architectures {
+				arches.Add(arch)
+			}
+		}
+		if arches.Size() == 1 {
+			cons.Arch = &arches.Values()[0]
+		}
+	}
```

Comment:
We could log out that we weren't able to set the constraint architecture because of ambiguity over the number of architectures.